gw_eccentricity/eccDefinition.py

```diff
@@ -1222,10 +1297,30 @@ def measure_ecc(self, tref_in=None, fref_in=None):
                 Measured mean anomaly at *tref_out*/*fref_out*. Same type as
                 *tref_out*/*fref_out*.
         """
+        # check that only one of tref_in or fref_in is provided
+        if (tref_in is not None) + (fref_in is not None) != 1:
+            raise KeyError("Exactly one of tref_in and fref_in"
+                           " should be specified.")
+        elif tref_in is not None:
+            # Identify whether the reference point is in time or frequency
+            self.domain = "time"
+            # Identify whether the reference point is scalar or array-like
+            self.ref_ndim = np.ndim(tref_in)
+            self.tref_in = np.atleast_1d(tref_in)
+        else:
+            self.domain = "frequency"
+            self.ref_ndim = np.ndim(fref_in)
+            self.fref_in = np.atleast_1d(fref_in)
         # Get the pericenters and apocenters
         pericenters = self.find_extrema("pericenters")
         original_pericenters = pericenters.copy()
-        self.check_num_extrema(pericenters, "pericenters")
+        # Check if there are a sufficient number of extrema. In cases where the
+        # waveform is long enough but the method fails to detect any extrema,
+        # it might be that the eccentricity is too small for the current method
+        # to detect it. See Fig.4 in arxiv.2302.11257. In such cases, we assume
+        # that the waveform is probably quasicircular.
```

Comment:
The last sentence might need to be updated?

Reply:
Done!